app/controllers/members_controller.rb

```diff
@@ -0,0 +1,70 @@
+class MembersController < ApplicationController
+  def new
+    query = "LEFT JOIN channel_users ON users.id = channel_users.user_id AND channel_users.channel_id = #{channel.id}"
+    @available_users = workspace.users
+                                .joins(query)
+                                .where('channel_users.id IS NULL')
+                                .order(:username)
```

Comment:
Never use string interpolation in SQL queries, especially if you are using it to insert params that come from the frontend. That would protect you from SQL injection.

```suggestion
    query = "LEFT JOIN channel_users ON users.id = channel_users.user_id AND channel_users.channel_id = ?"
    @available_users = workspace.users
                                .joins(query, channel.id)
                                .where('channel_users.id IS NULL')
                                .order(:username)
```

Reply:
Right! thanks!